test/server/routes/utils/authentication.test.js

```diff
@@ -0,0 +1,52 @@
+const Redis = require('ioredis');
+
+const logger = require(`${serverRoot}/config/winston`);
+const { sessionSignedOut, SIGNED_OUT_COOKIE } = require(`${testRoot}/server/_fakes/mocks/lib/session`);
+
+let app;
+
+const signedOutCookie = [`${process.env.COOKIE_NAME}=${SIGNED_OUT_COOKIE}`];
+
+describe('routes/applications.js', () => {
+  describe('signed out routes', () => {
+    beforeEach(done => {
+      sinon.reset();
+      sinon.restore();
+      sinon.stub(Redis.prototype, 'connect').returns(Promise.resolve());
+      sinon.stub(Redis.prototype, 'get').returns(Promise.resolve(sessionSignedOut));
+      app = require(`${serverRoot}/app`);
+
+      sinon.stub(logger, 'error').returns(true);
+      done();
+    });
+
+    afterEach(done => {
+      sinon.reset();
+      sinon.restore();
+      done();
+    });
+
+    const PROTECTED_PAGES = [
+      '/manage-applications',
+      '/manage-applications/add',
+      '/manage-applications/:appId/view/:env',
+      '/manage-applications/:appId/update/:env/:confirm',
+      '/manage-applications/:appId/delete',
+      '/manage-applications/:appId/api-key/add',
+      '/manage-applications/:appId/:keyType/:keyId/delete/:env',
+      '/manage-applications/:appId/api-key/update'
+    ];
```

Comment:
Given all pages in the scope of this app are protected could we more generally protect everything?

Reply:
Yes so everything is protected, but we want to hit each of these urls to make sure it is actually protected.

Reply:
Did you have to specify the urls to be protected or is this implicit within the session handler? This is an issue we currently have where our urls are not protected for `/manage-applications/*`